### ISSUE TITLE
Elsa 3.0.5 - Programmatic workflow with MassTransit MessageReceived activity not starting workflow

### ISSUE THREAD

**Issue by ckeller81** [elsa-workflows/elsa-core#4968]
**Summary:**
When using Elsa with Elsa.MassTransit and Elsa.MassTransit.RabbitMq and using the MessageReceived activity in a programmatically defined workflow as a start activity / trigger, the workflow is never started.

**Used Version**
All packages are from NuGet in the version 3.0.5, the example project does not have any other packages referenced.

**Example-Code:**
https://github.com/ckeller81/elsa-masstransit-test

**Reproduce steps:**
- Clone repo
- start RabbitMq from ./rabbitmq using 'docker-compose up'
- start project from ./src/ElsaMassTransitTest.csproj using 'dotnet run .\ElsaMassTransitTest.csproj'
- call http://localhost:5188/ to start a workflow from a queue-message
- call http://localhost:5188/simple to start a console-log workflow

**Expected behaviour:**
- the call to http://localhost:5188/simple writes "Hello world from simple workflow" to the console
- the call to http://localhost:5188/ writes "Hello world ... from Bus" to the console

**Actual behaviour:**
- :white_check_mark: the call to http://localhost:5188/simple writes "Hello world from simple workflow" to the console
- :x: the call to http://localhost:5188/ does **not** execute the workflow and does **not** write anything to the console

**Workaround**
Use elsa without MassTransit, consume the bus-message in a own consumer and start the workflow from this consumer.

**Thoughts**
After some hours of research, I found out that the queue-message is consumed correctly, but it seems that the workflow-start trigger is either not finding the workflow, or the bookmark. But this was a bit too deep in the package. 😄

**Comment by RyoukoKonpaku** [comment]
From my investigation on making this work on my end, it's because the `Type` property doesn't match with the incoming payload type which is used on the [WorkflowMessageConsumer](https://github.com/elsa-workflows/elsa-core/blob/662a0767269f7b623297daae94bea80249a6e2d7/src/modules/Elsa.MassTransit/Consumers/WorkflowMessageConsumer.cs#L30) to dispatch which activity gets triggered, so it never gets executed. Designer created triggers fine as it assigns the type property to the message type.

You can workaround it by overriding that property to match what it does when creating a trigger from the designer. 

```cs
new MessageReceived
{
    CanStartWorkflow = true,
    Type = ActivityTypeNameHelper.GenerateTypeName(typeof(YourMessageType)),
    MessageType = typeof(YourMessageType),
    Result = new(message)
}
```

**Comment by ckeller81** [comment]
@RyoukoKonpaku Thank you for the hint!
So it's more of a documentation issue, rather than a bug 😄 

Added the Type property, now it is working correctly.

**Comment by ckeller81** [comment]
@sfmskywalker is there any way to contribute to the documentation somehow? 😄

**Comment by sfmskywalker** [comment]
@ckeller81 For sure, thank you for asking! To contribute, please follow these steps:

1. Fork & clone the [elsa-documentation](https://github.com/elsa-workflows/elsa-documentation) repo.
2. Make the appropriate changes.
3. Submit a PR.

Notes:

- V3 documentation lives in the [V3 folder](https://github.com/elsa-workflows/elsa-documentation/tree/main/v3).
- We are migrating the current V3 documentation to the Writerside project which lives in the [writerside branch](https://github.com/elsa-workflows/elsa-documentation/tree/writerside/writerside). It's up to you to which version you would like to contribute, but ideally both versions get updated at the same time. Otherwise, we will make sure that your contributions are reflected in the Writerside branch down the road.

Please let me know if there's anything I can help with to get you going. Thanks for considering contributing to the documentation, we really could use some help!

**Comment by ckeller81** [comment]
Hey @hrb-amason 

The serialization issue you encounter is (as stated in #5210) because the System.Text.Json Serializer does not natively support serializing System.Type.

Mostly because of this issue (despite others), I always change the serializer to Newtonsoft.Json where I need JSON, also when using MassTransit and Elsa.

To change the serializer, you have to add the package MassTransit.Newtonsoft and set the Newtonsoft Serializer inside the RabbitMq configuration part (check the MassTransit documentation and examples for more info).

I would consider this, until #5210 is fixed.

Hope this helps.

**Comment by hrb-amason** [comment]
Hey @ckeller81

Thanks for the response. This was helpful, I was able to replace the serializer for mass transit. Is it possible to point me to sample on how to override the serializer in ELSA? 

I have ran into some other issues with the serializer and I haven't found a sample that replaces the serializer with Newtonsoft.

I tried creating my own implementation of [https://github.com/elsa-workflows/elsa-core/blob/main/src/modules/Elsa.Common/Contracts/IJsonSerializer.cs] but that hasn't worked.

Thanks again!
